pkg/istio-agent/xds_proxy_delta.go

```diff
@@ -296,6 +296,15 @@ func (p *XdsProxy) deltaRewriteAndForward(con *ProxyConnection, resp *discovery.
 		})
 		return
 	}
+
+	respResources := make([]*discovery.Resource, 0, len(resources))
```

Comment:
Should we add a test to validate that the revised config was indeed updated to the response before forwarding?

Reply:
I was thinking to add an integ test for applying WasmPlugin to the waypoint to make sure it has the correct config once this PR is merged - the problem I've mentioned above needs to be resolved firstly, otherwise the waypoint may not be ready if we have a WasmPlugin resource.

Reply:
Ok, thank you for clarifying!

Reply:
Can you test it via ut

Reply:
I think we will have coverage once https://github.com/istio/istio/pull/47969 is merged.